### ISSUE TITLE
changing metadata and access to data files

### ISSUE THREAD

**Issue by finn42** [capro-uio/nettskjemar#21]
I'm trying to use this library to retrieve data stored as zip files by an app using nettskjema. nettskjema_get_data() is producing a tibble of metadata that is 1) switching the user name and data file names and 2) not retrieving the linked files. Is there a way to prevent this renaming pattern? And is there yet an option to download the datafiles with this library?
<img width="1120" alt="Screen Shot 2021-10-18 at 18 30 56" src="https://user-images.githubusercontent.com/5003884/137772610-27ae05d3-bbb6-41c7-862a-88ed51a2cc82.png">

**Comment by Athanasiamo** [comment]
Thanks for the bug report and feature request. There must be something wrong in the coding of attachments (which I am assuming this field really is). So I'll have a peak at whats going on there.

In terms of downloading the file, I have not made code for this, but its a perfectly reasonable request. I'll have a peak at the API documentation and see what I can come up with. 

It wont be this week though, I as am busy with a conference.

**Comment by Athanasiamo** [comment]
I'll have a look later today, I think I can squeeze in some time. But how did you activate encryption of the attachments? When I test on a test form I made, I have no option to encrypt them, and thus I also cannot download them through the API :/

**Comment by Athanasiamo** [comment]
Hey!

I believe I have a set of functions for grabbing attachments now. I hope they are fairly well documented.
Test them out by installing from the development branch on GitHub 

```r
# install.packages("remotes")
remotes::install_github("LCBC-UiO/nettskjemar", ref = "dev")
```

Easiest solution is to use:
```r
nettskjema_get_form_attachments(form_id)
```

**Comment by finn42** [comment]
Oh cool! I haven't tried it yet as I fell back on accessing the API through python's request library, but this will be helpful for my colleague that mostly works in r. If you are curious about the python code, I've shared a demo notebook here https://github.com/finn42/PullingNettskjema/blob/main/PullingNettskjema.ipynb

**Comment by Athanasiamo** [comment]
base64 decoding was the little bit of code to figure out that made the magic happen. Super happy you flagged this for me to figure out, and also happy you found a way to do it in Python if that is your preferred language :) 

The more single users that use the API the better we can advocate for changes in the API for us. It was developed with server to server communication in mind, so for them its really novel to have single users wanting access to the nettskjema in this way.

**Comment by finn42** [comment]
Yeah I had a ticket open about the IP address restrictions, and have shared the same python jupyter notebook with them. Hopefully it will encourage them to consider how to make access a bit more practical. There was a lot of little details (like the base64 encoding) that could have been documented better.

**Comment by drmowinckels** [comment]
Hi!

We have new functions to download attachments, with some minimal documentation online.

https://www.capro.dev/nettskjemar/articles/attachments.html

I know we talked about this a long while ago, so I'm going to close the issue for now. Let me know if you still have problems.
